libs/contrib/Data/String/Extra.idr

```diff
@@ -89,10 +93,64 @@ index n str with (unpack str)
   index Z str | (x :: xs) = Just x
   index (S n) str | (x :: xs) = index n str | xs
 
+||| Produce a string by repeating the character `c` `n` times.
+public export
+replicate : (n : Nat) -> (c : Char) -> String
+replicate n c = pack $ replicate n c
+
+-- lines and unlines are cloned from Data.String for compatible reason, should be removed
```

Comment:
I don't think that applies here, so you can use `Data.String.lines` rather than cloning it.

Reply:
I should have read matt polzin's review, sorry